external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/CachedKafkaConsumer.scala

```diff
@@ -47,40 +51,191 @@ private[kafka010] case class CachedKafkaConsumer private(
 
   /** Iterator to the already fetch data */
   private var fetchedData = ju.Collections.emptyIterator[ConsumerRecord[Array[Byte], Array[Byte]]]
-  private var nextOffsetInFetchedData = -2L
+  private var nextOffsetInFetchedData = UNKNOWN_OFFSET
 
   /**
    * Get the record for the given offset, waiting up to timeout ms if IO is necessary.
```

Comment:
nit: update docs to clarify earlier that this may not return offset 
Get the record for the given offset if available. Otherwise it will either throw error (if failOnDataLoss = true), or return the next available offset within [offset, untilOffset].

Use @param to explain pollTimeoutMs and others in more detail